### ISSUE TITLE
How does Docker-ACI handle authentication against Azure Storage

### ISSUE THREAD

**Issue by timdagit** [docker/aci-integration-beta#34]
I've enjoyed playing with this today and was pleased by how easy it was to attach storage, but I wondered how my container was authenticating. I would normally expect to have to provide an access key, or configure a managed identity. It feels like the storage is trusting the container because it's in the same resource group but that feels odd. 

How does this work and will explicit control be provided in future?

**Comment by gtardif** [comment]
You're right when deploying containers to ACI you need to provide keys to the storage account, in the ACI API. 

The docker CLI will use your Azure login info (ie. `docker login azure`) to automatically fetch azure storage keys for the specified storage account, and add the required key in the payload sent to ACI for container deployment. 
It does not need specifically to be the same resource group, you need (as an azure user) to have access to the volume you are trying to mount in your ACI containers.

Basically, this automates the process described in [the aci volume docs](https://docs.microsoft.com/en-us/azure/container-instances/container-instances-volume-azure-files)
